mdtraj/core/selection.py

```diff
@@ -408,9 +409,8 @@ def __call__(self, selection):
         astnode = self.transformer.visit(deepcopy(parse_result[0].ast()))
 
         # Special check for a single literal
-        # TODO: These classes are deprecated since 3.8 and replaced with ast.Constant,
-        # but that substitution breaks behavior
-        if isinstance(astnode, (ast.Num, ast.Str)):
+
+        if isinstance(astnode, ast.Constant) and astnode.value not in {True, False, None}:
```

Comment:
I tried this a few months ago and I think something around here is where I got confused and gave up

Reply:
All `Num` and `Str` are `Constants`, but not all `Constants` are `Num` and `Str`. And so I had to filter out what comes out of L73 (True/False/None) vs. L79 (everything else). Everything else from L79 could eventually pass through other nodes (like Compare and what not, those will fail the first check), so those would be safe. But stuff that comes directly out of L79 could be single literal ('not') etc. which needs to choke and error out.

At least that's what i think is happening